### ISSUE TITLE
SonarTS doesn't understand newest TS lang features, concretely: optional chaining operator

### ISSUE THREAD

**Issue by abelepereira** [SonarSource/SonarTS#904]
**I want to report a bug.**

**SonarTS version:** 2.1<br>
**Node.js version:** 14<br>
**TypeScript version:** 3.8.3<br>

**Rule key:** typescript:S905: Expected an assignment or function call and instead saw an expression.

**Reproducer** 
```
foo?.bar(); // False-positive here: got "Expected an assignment or function call and instead saw an expression."
```

**Expected behavior**

Should simply understand the new (since TS v3.7) TS optional chaining operator (`?.`), and don't flag it as a bug.

**Comment by ed-graham** [comment]
Interestingly, this works for the SonarLint plug-in inside VS Code - no error is reported.  However, the SonarQube analysis fails with the error ``Failed to parse file ... at line ... Expression expected``.

**Comment by vilchik-elena** [comment]
Since SonarTS 2.0 typescript code is analyzed by SonarJS (SonarTS is basically empty plugin).
I will close this issue and review the problem in https://github.com/SonarSource/SonarJS/issues/1990
